Scripts/utils/read_csv_file.py

```diff
@@ -1,154 +1,42 @@
-        if zone_numbers is not None:
-            if not data.index.is_monotonic:
-                data.sort_index(inplace=True)
-                log.warn("File {} is not sorted in ascending order".format(path))
-            mapping = self.mapping
-            if mapping is not None:
-                log_path = mapping.name
-                if "total" in data.columns:
-                    # If file contains total and shares of total,
-                    # shares are aggregated as averages with total as weight
-                    data = data.groupby(mapping.zone_id).agg(avg, weights=data["total"])
-                else:
-                    share_cols = [col for col in data.columns
-                        if ("sh_" in col or "avg_" in col or "dummy" in col)]
-                    text_cols = data.columns[data.dtypes == object]
-                    funcs = dict.fromkeys(data.columns, "sum")
-                    for col in share_cols:
-                        funcs[col] = "mean"
-                    for col in text_cols:
-                        funcs[col] = "first"
-                    data = data.groupby(mapping).agg(funcs)
-                data.index = data.index.astype(int)
-            else:
-                log_path = path
-            if data.index.size != zone_numbers.size or (data.index != zone_numbers).any():
-                for i in data.index:
-                    if int(i) not in zone_numbers:
-                        msg = "Zone number {} from file {} not found in network".format(
-                            i, log_path)
-                        log.error(msg)
-                        raise IndexError(msg)
-                for i in zone_numbers:
-                    if i not in data.index:
-                        if mapping is not None and i in mapping.array:
-                            # If mapping is ok, then error must be in data file
-                            log_path = path
-                            i = mapping[mapping == i].index[0]
-                        msg = "Zone number {} not found in file {}".format(i, log_path)
-                        log.error(msg)
-                        raise IndexError(msg)
-                msg = "Zone numbers did not match for file {}".format(log_path)
+def read_other_data(path: Path, squeeze = False):
+    """Read data from space-separated file.
+
+    Parameters
+    ----------
+    path : Path
+        Path to the .tsv file
+    squeeze : bool (optional)
+        If the parsed data only contains one column and no header
+
+    Returns
+    -------
+    pandas.DataFrame
+    """
+    if not path.exists():
+        msg = f"Path {path} not found."
+        raise NameError(msg)
+    header = None if squeeze else "infer"
+    data: pandas.DataFrame = pandas.read_csv(
+        path, delim_whitespace=True, keep_default_na=False, 
+        na_values="", comment='#', header=header)
+    if "node_label" in data.columns:
```

Comment:
I'm not sure why this is here but I think "node_labels" should not be a hard-coded string in the data loader function. Maybe this could be an optional "index_column" argument that is then defined in the relevant configuration file or calling function.

Reply:
Yes, I agree. I put this back into the code to make it work as intended, but overall I have done no changes to this method. My plan is to completely reorganize these "other" data in an upcoming PR and put them all in one json-format file. After that we have the possibility to refer directly to one gpkg file (for zone data) and one json file (for data concerning travel costs) in the UI.